crates/sargon/src/signing/petition_types/petition_for_factors_types/petition_for_factors/petition_for_factors.rs

```diff
@@ -4,15 +4,15 @@ use crate::prelude::*;
 /// Petition of signatures from a factors list of an entity in a transaction.
 #[derive(Clone, PartialEq, Eq, derive_more::Debug)]
 #[debug("{}", self.debug_str())]
-pub(crate) struct PetitionForFactors {
+pub(crate) struct PetitionForFactors<S: Signable> {
```

Comment:
Looks like this can be made generic over `T: SignableID` only, not the whole `Signable` trait?